sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_directory_client.hpp

```diff
@@ -160,24 +160,65 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
+     * @return
+     * Azure::Core::Response<Models::SetDataLakeDirectoryAccessControlRecursiveSinglePageResult>
      * @remark This request is sent to dfs endpoint.
      */
-    Azure::Core::Response<Models::SetDataLakeDirectoryAccessControlRecursiveResult>
-    SetAccessControlRecursive(
-        Models::PathSetAccessControlRecursiveMode mode,
+    Azure::Core::Response<Models::SetDataLakeDirectoryAccessControlRecursiveSinglePageResult>
+    SetAccessControlRecursiveSinglePage(
         std::vector<Models::Acl> acls,
-        const SetDataLakeDirectoryAccessControlRecursiveOptions& options
-        = SetDataLakeDirectoryAccessControlRecursiveOptions()) const;
+        const SetDataLakeDirectoryAccessControlRecursiveSinglePageOptions& options
+        = SetDataLakeDirectoryAccessControlRecursiveSinglePageOptions()) const
+    {
+      return SetAccessControlRecursiveSinglePageInternal(
+          Models::PathSetAccessControlRecursiveMode::Set, acls, options);
+    }
+
+    /**
+     * @brief Updates POSIX access control rights on files and directories under given directory
+     * recursively.
+     * @param acls Updates POSIX access control rights on files and directories. Each access control
+     * entry (ACE) consists of a scope, a type, a user or group identifier, and permissions.
+     * @param options Optional parameters to set an access control recursively to the resource the
+     * directory points to.
+     * @return
+     * Azure::Core::Response<Models::UpdateDataLakeDirectoryAccessControlRecursiveSinglePageResult>
+     * @remark This request is sent to dfs endpoint.
+     */
+    Azure::Core::Response<Models::UpdateDataLakeDirectoryAccessControlRecursiveSinglePageResult>
+    UpdateAccessControlRecursiveSinglePage(
+        std::vector<Models::Acl> acls,
+        const UpdateDataLakeDirectoryAccessControlRecursiveSinglePageOptions& options
+        = UpdateDataLakeDirectoryAccessControlRecursiveSinglePageOptions()) const
+    {
+      return SetAccessControlRecursiveSinglePageInternal(
+          Models::PathSetAccessControlRecursiveMode::Modify, acls, options);
+    }
+
+    /**
+     * @brief Removes POSIX access control rights on files and directories under given directory
+     * recursively.
+     * @param acls Removes POSIX access control rights on files and directories. Each access control
+     * entry (ACE) consists of a scope, a type, a user or group identifier, and permissions.
+     * @param options Optional parameters to set an access control recursively to the resource the
+     * directory points to.
+     * @return
+     * Azure::Core::Response<Models::RemoveDataLakeDirectoryAccessControlRecursiveSinglePageResult>
+     * @remark This request is sent to dfs endpoint.
+     */
+    Azure::Core::Response<Models::RemoveDataLakeDirectoryAccessControlRecursiveSinglePageResult>
+    RemoveAccessControlRecursiveSinglePage(
+        std::vector<Models::Acl> acls,
```

Comment:
if `Remove`, why do we still need `acls`?

Reply:
Remove the specified ACLs.

Reply:
So you can remove some ACLs and keep others unchanged? You sure REST API supports this?

Reply:
I don't know but it's what .NET does.

Reply:
I was under the impression that we need to write unit test cases to cover these APIs.

Reply:
Added.